### PR TITLE
WIP: #379: #111: fix vscode to be portable/work in sandbox

### LINKED ISSUE #111: Assure portability of vsCode
Assure vsCode portability by setting the custom configuration for extension and home path.
Also you need to adapt the start script to check if no other instance is running, which is not started from the installation of the IDE as vsCode is working with shared application runtime, which would break with the project settings:

[vscode.txt](https://github.com/devonfw/devon-ide/files/3177462/vscode.txt)

### PR DESCRIPTION
**Pull request by hohwille**
PR trying to fix #111 and #379

### CODE REVIEW

File: scripts/src/main/resources/scripts/command/vscode [hunk 1]
```diff
@@ -55,7 +55,7 @@ function doAddPlugins() {
 
 function doAddPlugin() {
   doEchoAttention "Installing plugin (${1}) into VS code...\nPlease wait until VSCode has successfully installed all plugins.\nYou may then use or close VSCode as you like."
-  doRunCommand "'${VSCODE_HOME}/bin/code' --install-extension '${1}' &"
+  doRunVsCode --force --install-extension "${1}" &
```
Comment (maybeec) [inline scripts/src/main/resources/scripts/command/vscode]: I like the idea of installing a software package customized and updated by the software mechanics of the ide. Anyhow, just for curiosity as I see this method. Is there an additional setup description like for eclipse which plugins to install after provisioning of vscode?
If so, how do these two mechanisms relate to each other?

Comment (hohwille) [reply scripts/src/main/resources/scripts/command/vscode]: https://github.com/devonfw/ide/pull/381

Documentation for this PR is still missing. Further it is not working as expected. I need more help here.

Comment (hohwille) [reply scripts/src/main/resources/scripts/command/vscode]: Also refactored project import to be aligned with this and making code homogenous: https://github.com/devonfw/ide/pull/383

**Comment by sjimenez77** [comment]
Tested on my machine and worked fine. This is an image of my desktop:



The left VS Code instance is the one installed on my system. The right one is the one bundled on the devonfw ide. Both have different configuration and extensions installed. I reopened the both several times and the configuration was not changed or corrupted at all.

**Comment by hohwille** [comment]
CI failed due to network error:
```
[ERROR] Failed to execute goal org.codehaus.mojo:flatten-maven-plugin:1.1.0:flatten (flatten) on project devonfw-ide-doc: Execution flatten of goal org.codehaus.mojo:flatten-maven-plugin:1.1.0:flatten failed: Plugin org.codehaus.mojo:flatten-maven-plugin:1.1.0 or one of its dependencies could not be resolved: Could not transfer artifact org.apache.maven:maven-core:jar:3.2.5 from/to google-maven-central (https://maven-central.storage-download.googleapis.com/maven2/): GET request of: org/apache/maven/maven-core/3.2.5/maven-core-3.2.5.jar from google-maven-central failed: Connection reset -> [Help 1]
```
Will restart travis build

**Pull request merged**